### PR TITLE
Filter DotNetCore TargetFramework list selection to only include those with supported Sdks

### PR DESCRIPTION
**Pull request by iantoalms**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/983548

### CODE REVIEW

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectSupportedTargetFrameworks.cs [hunk 1]
```diff
@@ -97,6 +97,14 @@ public static IEnumerable<TargetFramework> GetNetCoreAppTargetFrameworks ()
 			}
 		}
 
+		public static IEnumerable<TargetFramework> GetNetCoreAppTargetFrameworksWithSdkSupport ()
+		{
+			foreach (var runtimeVersion in GetMajorRuntimeVersions ()) {
+				if (DotNetCoreSdk.Versions.Any(sdkVersion => runtimeVersion.Major == sdkVersion.Major && runtimeVersion.Minor == sdkVersion.Minor))
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectSupportedTargetFrameworks.cs]: Nit: indent
```suggestion
				if (DotNetCoreSdk.Versions.Any (sdkVersion => runtimeVersion.Major == sdkVersion.Major && runtimeVersion.Minor == sdkVersion.Minor))
```

**Comment by slluis** [comment]
@monojenkins backport release-8.3

**Comment by rodrmoya** [comment]
Code looks good and works, but the test failures seem to be legitimate

**Comment by rodrmoya** [comment]
@monojenkins backport release-8.3

**Comment by iantoalms** [comment]
@mrward Runtime version 2.0.3 shipped with Sdk 2.0.3 and Sdk 2.1.2 per this page: https://dotnet.microsoft.com/download/dotnet-core/2.0 All 2.0.4 and newer runtime releases shipped with 2.1.x Sdks.  There are similar major/minor version mismatches with 1.0 and 1.1 runtimes e.g 1.0.14 runtime in 1.1.12 Sdk - https://dotnet.microsoft.com/download/dotnet-core/1.0

**Comment by mrward** [comment]
@iantoalms Sorry my mistake. The old .NET Core 2.1.x sdk versioning confusion strikes again. Forgot they fixed all this confusion about the runtime version and the sdk version with .NET Core SDK 2.1.300 and later.

**Comment by slluis** [comment]
@monojenkins backport release-8.3-rtw

**Pull request merged**